parlai/agents/rag/rag.py

```diff
@@ -402,13 +403,48 @@ def update_state_dict(
             )
         return state_dict
 
+    def _should_override_dpr_model_weights(self, opt: Opt):
+        """
+        Determine if we need to override the DPR Model weights.
+
+        Under certain circumstances, one may wish to specify a different
+        `--dpr-model-file` for a pre-trained, RAG model. Thus, we additionally
+        check to make sure that the loaded DPR model weights are not overwritten
+        by the state loading.
+
+        NOTE: If `--model-file M` was trained with `--dpr-model-file D`, and
+        `--dpr-model-file D` is specified *after training* (i.e., in eval/interactive),
```

Comment:
Thought here - should there be an "--override-dpr-model-file-for-eval"?

(This was the initial case that triggered everything so...)

Reply:
i think this case i mention here was not what triggered things, but with the new implementation it's handled smoothly